File: CSharp/SpaceVIL/SpaceVIL/Core/ISize.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpaceVIL.Core
{
    /// <summary>
    /// An interface that describes the attributes of the size of a shape.
    /// <para/> This interface is part of SpaceVIL.Core.IBaseItem.
    /// </summary>
    public interface ISize
    {
        /// <summary>
        /// Method setting size of an item's shape.
        /// </summary>
        /// <param name="width">Width of a shape.</param>
        /// <param name="height">Height of a shape.</param>
        void SetSize(int width, int height);
        /// <summary>
        /// Method getting size of an item's shape.
        /// </summary>
        /// <returns>Size of the shape as SpaceVIL.Core.Size.</returns>
        Size GetSize();
        /// <summary>
        /// Method setting the minimum width limit. Actual width cannot be less than this limit.
        /// </summary>
        /// <param name="width"> Minimum width limit of the item. </param>
        void SetMinWidth(int width);
        /// <summary>
        /// Method setting item width. If the value is greater/less than the maximum/minimum 
        /// value of the width, then the width becomes equal to the maximum/minimum value.
        /// </summary>
        /// <param name="width">Width of the item.</param>
        void SetWidth(int width);
        /// <summary>
        /// Method setting the maximum width limit. Actual width cannot be greater than this limit.
        /// </summary>
        /// <param name="width">Maximum width limit of the item.</param>
        void SetMaxWidth(int width);
        /// <summary>
        /// Method getting the minimum width limit.
        /// </summary>
        /// <returns> Minimum width limit of the item. </returns>
        int GetMinWidth();
        /// <summary>
        /// Method fetting item width.
        /// </summary>
        /// <returns> Width of the item. </returns>
        int GetWidth();
        /// <summary>
        /// Method getting the maximum width limit.
        /// </summary>
        /// <returns> Maximum width limit of the item. </returns>
        int GetMaxWidth();
        /// <summary>
        ///  Method for setting the minimum height limit. Actual height cannot be less than this limit.
        /// </summary>
        /// <param name="height"> Minimum height limit of the item. </param>
        void SetMinHeight(int height);
        /// <summary>
        /// Method for setting item height. If the value is greater/less than the maximum/minimum 
        /// value of the height, then the height becomes equal to the maximum/minimum value.
        /// </summary>
        /// <param name="height"> Height of the item. </param>
        void SetHeight(int height);
        /// <summary>
        ///  Method for setting the maximum height limit. Actual height cannot be greater than this limit.
        /// </summary>
        /// <param name="height"> Maximum height limit of the item. </param>
        void SetMaxHeight(int height);
        /// <summary>
        /// Method for getting the minimum height limit.
        /// </summary>
        /// <returns> Minimum height limit of the item. </returns>
        int GetMinHeight();
        /// <summary>
        /// Method for getting item height.
        /// </summary>
        /// <returns> Height of the item. </returns>
        int GetHeight();
        /// <summary>
        /// Method for getting the maximum height limit.
        /// </summary>
        /// <returns> Maximum height limit of the item. </returns>
        int GetMaxHeight();
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/ISize.java
package com.spvessel.spacevil.Core;

/**
 * An interface that describes the attributes of the size of a shape.
 * <p> This interface is part of com.spvessel.spacevil.Core.IBaseItem.
 */
public interface ISize {
    /**
     * Method setting size of an item's shape.
     * @param width Width of a shape.
     * @param height Height of a shape.
     */
    public void setSize(int width, int height);

    /**
     * Method getting size of an item's shape.
     * @return Size of the shape as com.spvessel.spacevil.Core.Size.
     */
    public Size getSize();

    /**
     * Method setting the minimum width limit. Actual width cannot be less than this limit.
     * @param width Minimum width limit of the item.
     */
    public void setMinWidth(int width);

    /**
     * Method setting item width. If the value is greater/less than the maximum/minimum 
     * value of the width, then the width becomes equal to the maximum/minimum value.
     * @param width Width of the item.
     */
    public void setWidth(int width);

    /**
     * Method setting the maximum width limit. Actual width cannot be greater than this limit.
     * @param width Maximum width limit of the item.
     */
    public void setMaxWidth(int width);

    /**
     * Method getting the minimum width limit.
     * @return Minimum width limit of the item.
     */
    public int getMinWidth();

    /**
     * Method fetting item width.
     * @return Width of the item.
     */
    public int getWidth();

    /**
     * Method getting the maximum width limit.
     * @return Maximum width limit of the item.
     */
    public int getMaxWidth();

    /**
     * Method for setting the minimum height limit. Actual height cannot be less than this limit.
     * @param height Minimum height limit of the item.
     */
    public void setMinHeight(int height);

    /**
     * Method for setting item height. If the value is greater/less than the maximum/minimum 
     * value of the height, then the height becomes equal to the maximum/minimum value.
     * @param height Height of the item.
     */
    public void setHeight(int height);

    /**
     * Method for setting the maximum height limit. Actual height cannot be greater than this limit.
     * @param height Maximum height limit of the item.
     */
    public void setMaxHeight(int height);

    /**
     * Method for getting the minimum height limit.
     * @return Minimum height limit of the item.
     */
    public int getMinHeight();

    /**
     * Method for getting item height.
     * @return Height of the item.
     */
    public int getHeight();

    /**
     * Method for getting the maximum height limit.
     * @return Maximum height limit of the item.
     */
    public int getMaxHeight();
}